crates/sui-graphql-client/src/lib.rs

```diff
@@ -602,6 +602,57 @@ impl Client {
     // Transaction API
     // ===========================================================================
 
+    /// Dry run a transaction and return the transaction.
+    ///
+    /// `txBytes` is either a TransactionData struct or a [`TransactionKind`] struct, BCS-encoded
+    /// and then Base64-encoded. The expected type is controlled by the presence or absence of
+    /// `txMeta`:
+    ///
+    ///  - if present, txBytes is assumed to be a [`TransactionKind`],
+    ///  - if absent, txBytes is assumed to be [`Transaction`].
+    ///
+    /// `skipChecks` optional flag to disable the usual verification checks that prevent access to
+    /// objects that are owned by addresses other than the sender, and calling non-public,
+    /// non-entry functions, and some other checks. Defaults to false.
```

Comment:
Do we need to have two dry_run apis? This doc comment doesn't match the function signature now that it takes &Transaction...

Reply:
I don't follow the `two dry_run apis`? Were you referring to the usage of `txMeta` to determine the type of the transaction to dry run?

I assume we can always just use `Transaction`, and do not pass the `txMeta` argument. I am not sure what was the rationale behind having this two-version dry-run in GraphQL.

Reply:
your comment says that the metadata only is used if a TransactionKind is sent, so a good api would split these two cases into separate apis, a Kind with metadata and a Transaction only api

Reply:
Ah I see what you mean. Thanks for that!

Reply:
Refactored